### PR TITLE
Remove conditon in cuda kernel with template

### PR DESCRIPTION
**Pull request by yufenglee**
**Description**: Describe your changes.

**Motivation and Context**
- Why is this change required? What problem does it solve?
- If it fixes an open issue, please link to the issue here.

### CODE REVIEW

File: onnxruntime/contrib_ops/cuda/bert/skip_layer_norm_impl.cu [hunk 1]
```diff
@@ -69,44 +79,72 @@ __global__ void SkipLayerNormKernel(
...
   assert(n % ld == 0);
   const int grid_size = n / ld;
 
   if (ld <= 32) {
-    constexpr int block_size = 32;
-    SkipLayerNormKernelSmall<T, block_size>
-        <<<grid_size, block_size, 0, stream>>>(ld, input, skip, beta, gamma, bias, output);
+    InvokeSkipLayerNormKernelSmall(32, bias)
   } else if (ld <= 128) {
     constexpr int block_size = 128;
```
Comment (tianleiwu) [inline onnxruntime/contrib_ops/cuda/bert/skip_layer_norm_impl.cu]: The following lines can be removed:
   constexpr int block_size = 128; 
   constexpr int block_size = 384; 
and apply similar change for block_size=256

**Comment by yufenglee** [comment]
> What's the performance impact of this change?
> Use template will increase the binary size. If there is no notable gain in performance, I would prefer original code which is simpler (less code).

The condition in the kernel will be optimized out with this change. As for the binary size, the main concern is on the CPU provider.

**Pull request closed without merging**